2016/exercises/final_project/Alba-Petar-Rebecca-Simon/PolytopeProject/Point.cpp

```diff
@@ -0,0 +1,84 @@
+//
+//  Point.cpp
+//  PolytopeProject
+//
+
+
+#include "Point.hpp"
+
+
+Point::Point(vector<mpq_class> coord,bool sign):sign(sign){
+    for(int i = 0; i<coord.size(); i++){
+        mpq_class rational = coord[i];
+        rational.canonicalize();
+        coordinates.push_back(rational);
+    }
+}
+
+Point::Point(vector<double> coord, bool sign):sign(sign){
+    for(double dob : coord){
+        mpq_class q(dob);
+        coordinates.push_back(q);
+    }
+}
+
```

Comment:
If you introduce the invariant that upon construction of a `vector<mpq_class>` all entries are canocnicalized, you can save a lot of calls to `canonicalize()`, and replace the two versions here by a templated version